.azure-pipelines/ci-build.yml

```diff
@@ -198,12 +202,29 @@ extends:
             targetFolder: $(Build.ArtifactStagingDirectory)/Nugets
             sourceFolder: $(Build.ArtifactStagingDirectory)
             content: '*.nupkg'
-            
+
+        # Copy repository files to be used in the deploy stage
+        - task: CopyFiles@2
+          displayName: 'Copy repository files for deploy stage'
+          inputs:
+            SourceFolder: '$(Build.SourcesDirectory)'
+            Contents: |
```

Comment:
have you considered this option instead of copying the whole repo which is effectively working around the security constraints?
https://stackoverflow.com/questions/66468815/how-to-publish-docker-image-as-an-artifact-in-azure-devops